### ISSUE TITLE
Regression between 5.4.1 and 5.7+ using sof: no more audio output using sof

### ISSUE THREAD

**Issue by gled-rs** [thesofproject/linux#2427]
Hi,

On kernel 5.4.1 ( last known good and tested ), everything was working properly, dmic and audio output on the speakers.

Updating to kernel 5.7+ ( tested up to 5.8.7 ):

- Without any mod params, dmic works perfectly, but there's no more audio output, even if the card is still detected properly, and seems to be working, jack plug detection works when plugging a headset, pulseaudio indicates something is playing, just no audio is getting out of the speakers nor the headset when plugged.
[sof_alsa_nomodparams.txt](https://github.com/thesofproject/linux/files/5203303/sof_alsa_nomodparams.txt)

- Using legacy driver ( dsp_driver=1 ), audio output works perfectly but no dmic.
[sof_alsa_dsp_driver1.txt](https://github.com/thesofproject/linux/files/5203304/sof_alsa_dsp_driver1.txt)

- Using new driver ( dsp_driver=2 ), no card is detected at all
[sof_alsa_dsp_driver2.txt](https://github.com/thesofproject/linux/files/5203305/sof_alsa_dsp_driver2.txt)

Let me know if there's anything more I can provide to help diagnose the source of the regression.

**Comment by plbossart** [comment]
the option dsp_driver=2 should not work here, it's irrelevant.

The only options that are relevant are dsp_driver=0 (or not option) -> automatic detection and dsp_driver=1 to force the legacy to be used.

We use exactly the same codec configuration code between legacy and SOF driver, so we need to figure out if this is a PulseAudio issue or if you can get basic "speaker-test -Dhw:0,0 -c2 -r48000" to work for your speaker output?
If the streaming works with speaker-test, check with alsamixer if one switch is off maybe?

If you get sound output at the hw: device level, then most likely it's a pulseaudio/UCM installation issue.

**Comment by gled-rs** [comment]
thank you for the note on dsp_driver=2, that explains things.

Currently running with dsp_driver=0, since with dsp_driver=1 there's no dmic.

 "speaker-test -Dhw:0,0 -c2 -r48000" outputs no audio. no errors, it seems to be running properly. 

Everything in alsamixer seems to be on, and volume at max. I tried both automute on and off.

![Screenshot_2020-09-10_10-41-09](https://user-images.githubusercontent.com/5700851/92774288-2b724b80-f352-11ea-9f29-885f28d48339.png)

[edit]
speaker-test does output audio when dsp_driver=1

**Comment by plbossart** [comment]
@kv2019i does this ring a bell? I vaguely remember one user setting an additional smixer that wasn't set correctly, but other than that the basic playback should use the same settings as the legacy audio.

**Comment by juimonen** [comment]
@plbossart just a hunch, but can we somehow disable sof/codec power management easily from config and see what happens?

**Comment by kv2019i** [comment]
@plbossart @juimonen  Hmm, indeed it's the same codec driver. The one big remaining different is indeed codec runtime handling. 

@gled-rs One suspect is this:
3a24f135e6cc58c2dc042dde44031ec31b706ce5 ASoC: intel/skl/hda - set autosuspend timeout for hda codecs
... but it's not included in 5.7 kernel, so I don't think that explains.

Another curious in the logs is that in case SOF/DSP is enabled, you have 'Master Playback Switch' muted. This seems really strange, but easy to test by doing the following doing playback test:

amixer cset iface=MIXER,name='Master Playback Switch' on

... if this does help, then Arch Linux is doing something strange in user-space. The UCM files we ship upstream (alsa-ucm) for UCM certainly set this control correctly.

**Comment by gled-rs** [comment]
Hi guys,

Thank you for looking into this.

I just checked, UCM files are vanilla from **https://www.alsa-project.org/files/pub/lib/alsa-ucm-conf-1.2.3.tar.bz2** apparently.

I did "amixer cset iface=MIXER,name='Master Playback Switch' on" and then a playback test, and audio output works !

Sorry I missed that and for the noise, clearly something in userspace that muted that switch or set it muted by default during the kernel update.

Thank you again :)
